### PR TITLE
Early termination

### PR DESCRIPTION
**Pull request by leelaprabhu**

### CODE REVIEW

File: deepchem/models/tensorgraph/tensor_graph.py [hunk 1]
```diff
@@ -106,7 +106,26 @@ def _add_layer(self, layer):
     for in_layer in layer.in_layers:
       self._add_layer(in_layer)
       self.nxgraph.add_edge(in_layer.name, layer.name)
-
+  
+  def _eval_valid(self, sess, valid_fdg):
```
Comment (lilleswing) [inline deepchem/models/tensorgraph/tensor_graph.py:110]: why do we need this vs just using predict or predict proba?

File: deepchem/models/tensorgraph/tensor_graph.py [hunk 2]
```diff
@@ -367,6 +395,10 @@ def _install_queue(self):
   def set_loss(self, layer):
     self._add_layer(layer)
     self.loss = layer
+
```
Comment (lilleswing) [inline deepchem/models/tensorgraph/tensor_graph.py:398]: cost should be passed into fit

File: deepchem/models/tensorgraph/tensor_graph.py (hunk 1 above)
Comment (leelaprabhu) [reply deepchem/models/tensorgraph/tensor_graph.py:110]: Oh yeah that's much neater, I'll try that now. I just thought of this because it uses the loss directly.
